clients/src/main/java/org/apache/kafka/common/MetricNameTemplate.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.kafka.common.utils.Utils;
+
+/**
+ *         A template for a MetricName. It contains a name, group, and
+ *         description, as well as all the tags that will be used to create the
+ *         mBean name. Tag values are omitted from the template, but are filled
+ *         in at runtime with their specified values.
+ */
+public class MetricNameTemplate {
```

Comment:
There's some background discussion on this, in https://issues.apache.org/jira/browse/KAFKA-3480